tests/testthat/test-trailing_blank_lines_linter.R

```diff
@@ -89,10 +105,64 @@ test_that("returns the correct linting", {
     ),
     file = tmp5
   )
-  expect_lint(content = NULL, file = tmp5, list(
-    message = msg2,
-    line_number = 10L,
-    # We can't get 4 here because the line is NA-masked in get_source_expressions(), so no line length info exists.
-    column_number = 1L
-  ), linter)
+  expect_lint(
+    file = tmp5,
+    checks = list(
+      message = rex::rex("Missing terminal newline."),
+      line_number = 10L,
+      # We can't get 4 here because the line is NA-masked in get_source_expressions(), so no line length info exists.
+      column_number = 1L
+    ),
+    linters = linter
+  )
+})
+
+test_that("blank lines in knitr chunks produce lints", {
+  linter <- trailing_blank_lines_linter()
+
+  tmp6 <- withr::local_tempfile(
```

Comment:
The actual string written to file for this one will look very odd compared to what you would guess at a glance. Is trim_some() needed?

Reply:
Indeed, it is needed. I have made this change now